test/functional/normal/tab_spec.lua

```diff
@@ -0,0 +1,230 @@
+       .. 'Tab page 5\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to next tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the next (first) tab
+    feed('gt')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '>   [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to last tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the next (first) tab
+    feed('gt')
+    feed(':tablast<cr>')
+    -- Switch to the earlier (second) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '>   [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to previous tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the previous (third) tab
+    feed('gT')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '>   [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after switching to first tab', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the previous (third) tab
+    feed('gT')
+    -- Switch to the first tab
+    feed(':tabfirst<cr>')
+    -- Switch to the earlier (third) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
+       .. '\n'
+       .. 'Tab page 1\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 2\n'
+       .. '    [No Name]\n'
+       .. 'Tab page 3\n'
+       .. '>   [No Name]\n'
+       .. 'Tab page 4\n'
+       .. '    [No Name]',
+       redir_exec('tabs')
+    )
+  end)
+
+  it('switches to earlier after numbered tab switch', function()
+    -- Add three tabs for a total of four
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    feed(':tabnew<cr>')
+    -- Switch to the second tab
+    feed('2gt')
+    -- Switch to the earlier (fourth) tab
+    feed(':tabantecedent<cr>')
+
+    eq(   '\n'
```

Comment:
can use a Lua literal string instead of concatenating a bunch of strings. We also have `helpers.dedent()` so you can indent it nicely.

```
eq([[
  Tab page 1
    [No Name]
  Tab page 2
    ...
]]
```

Reply:
Fixed to use `eq(dedect([[...`.  One unpleasant remaining thing is that since these strings all end with `]`, it looks like a separator character is required to distinguish that final `]` from the first `]` of the sigil indicating the end of the string literal `]]`; if that's right, the consequence is that the separator has to be appended to the other string being compared.  For now, `\n` is being used as the separator.  If there's a way to avoid this, let's do it.

Reply:
@butwerenotthereyet to avoid such `]]` ambiguity, Lua allows any number of `=` between the brackets, thus `[=[ ... ]=]` instead of `[[ ... ]]`

Reply:
That's perfect!  Fixed.